webapp/components/interop.html

```diff
@@ -0,0 +1,417 @@
+    note that tested Edge and Safari are not pre-release versions (<a
+          href="https://github.com/w3c/wptdashboard/issues/109">#109</a>, <a
+          href="https://github.com/w3c/wptdashboard/issues/110">#110</a>).
+  </section>
+
+  <section class="runs">
+    <table>
+      <thead>
+      <tr>
+        <template is="dom-repeat" items="{{ allMetrics.metadata.test_runs }}"
+                  as="testRun">
+          <th>
+            <test-run test-run="[[testRun]]"></test-run>
+          </th>
+        </template>
+      </tr>
+      </head>
+    </table>
+  </section>
+
+  <section class="search">
+    <template is="dom-if" if="{{ !pathIsATestFile }}">
+      <input value="{{ query::input }}"
+             class="query"
+             placeholder="{{ queryPlaceholder }}"
+             data-foo="bar">
+    </template>
+    <div class="path">
+      <a href="/interop/" on-click="_navigate">WPT</a>
+      <template is="dom-repeat" items="{{ _splitIntoLinkedParts(path) }}" as="part">
+        <span class="path-separator">/</span>
+        <a href="/interop{{ part.path }}"  on-click="_navigate">{{ part.name }}</a>
+      </template>
+    </div>
+  </section>
+
+  <template is="dom-if" if="{{ pathIsATestFile }}">
+    <div class="links">
+      <ul>
+        <li>
+          <a href$="https://github.com/w3c/web-platform-tests/blob/master[[path]]"
+             target="_blank">View source on GitHub</a></li>
+        <li><a href$="https://w3c-test.org[[path]]" target="_blank">Run in your
+          browser on w3c-test.org</a></li>
+      </ul>
+    </div>
+  </template>
+
+  <table>
+    <thead>
+      <tr>
+        <th>Passing:</th>
+        <template is="dom-repeat" items="{{ thLabels }}" as="label">
+          <th>{{ label }}</th>
+        </template>
+      </tr>
+    </thead>
+    <tbody>
+      <template is="dom-repeat" items="{{ metrics.data }}" as="node">
+        <tr>
+          <td>
+            <template is="dom-if" if="{{ !_computePathIsATestFile(node.dir) }}">
+              <b><a href="/interop{{ node.dir }}" on-click="_navigate">{{ _relPath(node.dir) }}</a></b>
+            </template>
+            <template is="dom-if" if="{{ _computePathIsATestFile(node.dir) }}">
+              <a href="/interop{{ node.dir }}" on-click="_navigate">{{ _relPath(node.dir) }}</a>
+            </template>
+          </td>
+
+          <template is="dom-repeat" items="{{node.pass_rates}}" as="passRate">
+            <td style="{{ _passRateStyle(node, passRate) }}">{{ passRate }} / {{ node.total }}</td>
+          </template>
+        </tr>
+      </template>
+    </tbody>
+  </table>
+
+</template>
+
+<script>
+  'use strict';
+  const urlToMetricsPath = url => url.pathname.replace(/\/interop(.+)(\/)?$/, '$1');
+  const urlToQuery = url => {
+    // Drop initial "?" from url.search.
+    const search = url.search;
+    if (!search) {
+      return '';
+    }
+    const match = search.match(/[^a-zA-Z0-9%]q=([^&]+)/);
```

Comment:
I would like to see a comprehensive data set containing example urls that are expected to match.

Reply:
@mdittmer